src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -906,11 +1019,13 @@ def get_connected_passed_kwargs(prefix):
                 f"Pipeline {pipeline_class} expected {expected_modules}, but only {passed_modules} were passed."
             )
 
-        # 8. Instantiate the pipeline
+        # 10. Instantiate the pipeline
         model = pipeline_class(**init_kwargs)
 
-        # 9. Save where the model was instantiated from
+        # 11. Save where the model was instantiated from
         model.register_to_config(_name_or_path=pretrained_model_name_or_path)
+        if device_map is not None and isinstance(device_map, str) and device_map in ["balanced"]:
```

Comment:
To simplify a bit if we do the check I proposed above.

Reply:
Done in 25d04b82.

Reply:
thanks for the fix !